### PR TITLE
Update doc comment for json writer's requirement of slack space and add to  other APIs that need it.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Follow-up to https://github.com/Azure/azure-sdk-for-c/pull/1972

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

File: sdk/inc/azure/core/az_json.h [hunk 1]
```diff
@@ -420,20 +419,19 @@ AZ_NODISCARD az_result az_json_writer_append_string(az_json_writer* ref_json_wri
  * is, without any formatting or spacing changes. No modifications are made to this text, including
  * escaping.
  *
+ * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
+ * sufficient space, note that the JSON writer requires at least 64-bytes of slack within the
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_json.h:423]:
```suggestion
 * sufficient space, note that the JSON writer requires at least 64 bytes of slack within the
```

File: sdk/inc/azure/core/az_json.h [hunk 2]
```diff
@@ -390,18 +390,17 @@ az_json_writer_get_bytes_used_in_destination(az_json_writer const* json_writer)
...
- * because it tries to write in 64 byte chunks (10 character * 6 if all need to be escaped into the
- * unicode form).
- *
  * @param[in,out] ref_json_writer A pointer to an #az_json_writer instance containing the buffer to
  * append the string value to.
  * @param[in] value The UTF-8 encoded value to be written as a JSON string. The value is escaped
  * before writing.
  *
+ * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
+ * sufficient space, note that the JSON writer requires at least 64-bytes of slack within the
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_json.h:399]:
```suggestion
 * sufficient space, note that the JSON writer requires at least 64 bytes of slack within the
```

File: sdk/inc/azure/core/az_json.h [hunk 3]
```diff
@@ -456,6 +454,12 @@ az_json_writer_append_json_text(az_json_writer* ref_json_writer, az_span json_te
  * @param[in] name The UTF-8 encoded property name of the JSON value to be written. The name is
  * escaped before writing.
  *
+ * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
+ * sufficient space, note that the JSON writer requires at least 64-bytes of slack within the
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_json.h:458]:
```suggestion
 * sufficient space, note that the JSON writer requires at least 64 bytes of slack within the
```

File: sdk/inc/azure/core/az_json.h [hunk 4]
```diff
@@ -484,11 +488,10 @@ AZ_NODISCARD az_result az_json_writer_append_bool(az_json_writer* ref_json_write
  * @param[in] value The value to be written as a JSON number.
  *
  * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
- * theoretically space, note that the JSON writer requires at least 64-bytes of slack within the
+ * sufficient space, note that the JSON writer requires at least 64-bytes of slack within the
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_json.h:491]:
```suggestion
 * sufficient space, note that the JSON writer requires at least 64 bytes of slack within the
```

File: sdk/inc/azure/core/az_json.h [hunk 5]
```diff
@@ -506,11 +509,10 @@ AZ_NODISCARD az_result az_json_writer_append_int32(az_json_writer* ref_json_writ
  * point and truncate the rest.
  *
  * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
- * theoretically space, note that the JSON writer requires at least 64-bytes of slack within the
+ * sufficient space, note that the JSON writer requires at least 64-bytes of slack within the
```
Comment (antkmsft) [inline sdk/inc/azure/core/az_json.h:512]:
```suggestion
 * sufficient space, note that the JSON writer requires at least 64 bytes of slack within the
```

**Comment by ahsonkhan** [comment]
@danewalton any thoughts?

**Pull request merged**